### PR TITLE
Anonymous records

### LINKED ISSUE #11764: Missing documentation on F# Anonymous Records
I realize F# 4.6 is quite new but it would be great to have the documentation for new F# features to stay updated along with the additions to the language such as Anonymous Records. Having this documentation up-to-date would serve as a better reference in places like Stack Overflow or other places where we to point somebody to official documentation for the new feature. Other than GitHub issues and some blog posts I don't see anything official describing Anonymous Records and their prospective uses.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe1bc735-dea5-eaf1-8adf-6fa03a43a88e
* Version Independent ID: 76d64ebf-a460-3e86-898f-c31c6c70e354
* Content: [Records - F#](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/records#feedback)
* Content Source: [docs/fsharp/language-reference/records.md](https://github.com/dotnet/docs/blob/master/docs/fsharp/language-reference/records.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **phcart**

### PR DESCRIPTION
**Pull request by cartermp**
Fixes https://github.com/dotnet/docs/issues/11764

Checklist:

- [x] Syntax
- [x] Fundamental properties (serializable, can cross assembly boundaries, nominal, etc.)
- [x] Basic construction
- [x] Using as type argument
- [x] Copy and update
- [x] Augmenting existing data
- [x] Limitations

### CODE REVIEW

File: docs/fsharp/language-reference/anonymous-records.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+### Limitations with pattern matching
+
+Anonymous records do not support pattern matching, unlike named records. This is for 3 reasons:
+
+1. A pattern would have to account for every field of an anonymous record, unlike named record types. This is because anonymous records do not support structural subtyping – they are nominal types.
+2. Because of (1), there is no ability to have additional patterns in a pattern match expression, as each distinct pattern would imply a different anonymous record type.
+3. Because of (3), any anonymous record pattern would be more verbose than the use of “dot” notation.
+
+There is an open language suggestion to [allow mattern matching in limited contexts](https://github.com/fsharp/fslang-suggestions/issues/713).
```
Comment (purkhusid) [inline docs/fsharp/language-reference/anonymous-records.md]:
```suggestion
There is an open language suggestion to [allow pattern matching in limited contexts](https://github.com/fsharp/fslang-suggestions/issues/713).
```

**Review by BillWagner — APPROVED** [review]
This is a great explanation @cartermp 

I found a few nits, and had one suggestion where I had trouble understanding your intent. Consider the way I reworded that section. Then, you can :shipit: when ready.

**Comment by cartermp** [comment]
@BillWagner @mairaw I can't seem to run Acrolinx on my mac at home. Do I need to be on corpnet?

**Comment by cartermp** [comment]
I signed in. Weird.

**Pull request merged**